### ISSUE TITLE
Parens in return type template are dropped

### ISSUE THREAD

**Issue by nilsleiffischer** [doxygen/doxygen#8465]
**Describe the bug**

This C++ function is not parsed correctly by Doxygen:

```cpp
/// One of the opening parens in the return type is dropped by Doxygen.
template <typename T>
MyTemplate<decltype(std::declval<T>())> broken_func();
```

I created a minimal working example to reproduce this issue here:

- MWE: [issue_doxygen_templates.zip](https://github.com/doxygen/doxygen/files/6221766/issue_doxygen_templates.zip) or
https://github.com/nilsleiffischer/sphinx-bugs/tree/master/issue_doxygen_templates

Running `doxygen Doxyfile` in this directory produces XML and HTML output. Here's a screenshot from the HTML output that shows the issue:

<img width="741" alt="Bildschirmfoto 2021-03-29 um 14 41 44" src="https://user-images.githubusercontent.com/746230/112838555-76a93d00-909d-11eb-8d79-bdceac8e5f80.png">

**Expected behavior**

The return type should read `MyTemplate<decltype(std::declval<T>())>` (note the missing parenthesis in the screenshot above).

**Version**

- Doxygen version 1.9.1
- macOS 11.2.3

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8466

**Comment by nilsleiffischer** [comment]
@albert-github wow that was fast, I'm impressed.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by nilsleiffischer** [comment]
Thanks for fixing this so fast!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
